### PR TITLE
Add 'reset' method

### LINKED ISSUE #36: Purge state on logout?
When users log out of our app, I clear fields like email address by calling `identify` with `null` fields. But upon login it seems that new user data is being associated with the _old_ survey records in Integrate's backend. Example:

1. User 1 login => `.identify({email: user1@gmail.com})`
2. User 1 submit `survey1`
3. User 1 logout => `.identify({email: null})`
4. User 2 login => `.identify({email: user2@gmail.com})`
5. User 2 submit `survey2`

The Integrate backend now shows `survey1` and `survey2` both associated with `user2`. Interestingly this seems to also happen even through full app delete/reset?

There is a fair bit of phone device reuse in our user population, so this use case is important to cover to avoid misattributing survey data.

### PR DESCRIPTION
**Pull request by msingleton**
Add a 'reset' method that clears all data from local storage and the local redux store. This is useful when logging out a user to ensure their data isn't associated with the next logged in user. Fixes #36

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -222,6 +222,22 @@ If you need access to the user's responses on the client, you can use the `onRes
     }
 ```
 
+## Clearing data
+
+To clear all data Iterate has stored (user api key, any user properties stored by calling the `identify` method, etc) call the `reset` method. This is commonly called when you log a user out of your app.
+
+```JSX
+    const App: () => JSX.Element = () => {
```
Comment (lankybutmacho) [inline README.md:230]: I think you can cut out the () => JSX.Element type here - not the most common type used I think and might be distracting/confusing

File: src/redux.tsx [hunk 2]
```diff
@@ -107,6 +111,12 @@ export const reducer = (state: State = INITIAL_STATE, action: Action) => {
         showSurvey: false,
         showPrompt: false,
       };
+    case 'RESET':
+      // We don't need to reset the company API key, so we'll
+      // reset to the initial state and cherry pick that
+      const updatedState: State = { ...INITIAL_STATE };
```
Comment (lankybutmacho) [inline src/redux.tsx:117]: Style choice up to you but you could do `const updatedState: State = { ...INITIAL_STATE, companyAuthToken: state.companyAuthToken };` and remove the next line

**Review by lankybutmacho — APPROVED** [review]
lgtm, 2 small notes

File: README.md (hunk 1 above)
Comment (msingleton) [reply README.md:230]: yea good call

File: src/redux.tsx (hunk 2 above)
Comment (msingleton) [reply src/redux.tsx]: yea that's better

**Pull request merged**